src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts

```diff
@@ -57,28 +64,36 @@ export class DecorationAddon extends Disposable implements ITerminalAddon {
 		this._register(this._contextMenuService.onDidShowContextMenu(() => this._contextMenuVisible = true));
 		this._register(this._contextMenuService.onDidHideContextMenu(() => this._contextMenuVisible = false));
 		this._hoverDelayer = this._register(new Delayer(this._configurationService.getValue('workbench.hover.delay')));
+
 		this._configurationService.onDidChangeConfiguration(e => {
 			if (e.affectsConfiguration(TerminalSettingId.ShellIntegrationDecorationIcon) ||
 				e.affectsConfiguration(TerminalSettingId.ShellIntegrationDecorationIconSuccess) ||
 				e.affectsConfiguration(TerminalSettingId.ShellIntegrationDecorationIconError)) {
-				if (this._placeholderDecoration?.element) {
-					this._applyStyles(this._placeholderDecoration.element);
-				}
-				for (const decoration of this._decorations) {
-					if (decoration[1].decoration?.element) {
-						this._applyStyles(decoration[1].decoration.element, decoration[1].exitCode);
-					}
-				}
-			} else if (e.affectsConfiguration(TerminalSettingId.ShellIntegrationDecorationsEnabled)) {
-				if (!this._configurationService.getValue(TerminalSettingId.ShellIntegrationDecorationsEnabled)) {
-					this._commandStartedListener?.dispose();
-					this._commandFinishedListener?.dispose();
-					this._clearDecorations();
-				}
+				this._refreshClasses();
+			} else if (e.affectsConfiguration(TerminalSettingId.FontSize)) {
+				this.refreshLayouts();
```

Comment:
We're only taking font size into account here but line height is also important. With lineHeight = 2:

![image](https://user-images.githubusercontent.com/2193314/155774187-680621f9-710f-409a-9c4a-61ee8286aa76.png)

This is the difference between char height vs cell height in xterm.

Reply:
![Screen Shot 2022-02-25 at 1 22 06 PM](https://user-images.githubusercontent.com/29464607/155775344-f5fc4b82-45e8-4855-aff4-ba94fc4a21d7.png)